### DISCUSSION TITLE
guide-R/analyses/effets-marginaux

### DISCUSSION THREAD

**Discussion by giscus** [larmarange/guide-R#14] [Forum]
# guide-R/analyses/effets-marginaux

https://larmarange.github.io/guide-R/analyses/effets-marginaux.html

**Comment by mthevenin** [comment]
Bonjour,
Quelques remarques avec R.

- Depuis 2021 il y a un nouveau pkg "marginaleffects" (VA.Bundock), un peu plus ambitieux que "margins".  
Vous dites que le pkg margins est l'équivalent de Stata. Il a nettement moins d'ambitieux, et c'est franchement pas plus mal, tant le margins de Stata peut se transformer en un belle usine à gaz avec des éléments statistiquement contestables (au secours le calcul de la variances de probabilités ajustées après l'estimation d'un modèle logit, probit ou poisson)

- Vous présentez des AME, très bonne idée, mais la définition que vous en donnez n'est pas pas celle qui sera estimée par défaut par margins de R: la valeur des autres variables n'est pas fixée à leur niveau moyen mais à leur valeur observée. Le pkg n'a pas implémenté cette option: "Stata’s atmeans argument is not implemented in margins() for various reasons, including because it is possible to achieve the effect manually through an operation like data$var <- mean(data$var, na.rm = TRUE) and passing the modified data frame to margins(x, data = data). Il y a plutôt consensus pour les variables discrètes à rester sur les valeurs observées, donc sur l'estimation implémentée par défaut par le pkg.  
Contrairement aux MEM (marginal effect at means), l'AME d'une variable est la simple moyenne d'effets marginaux calculés au niveau individuel. Ceux ci sont obtenus par une opération de contrefactualisation individuelle de la caractéristique pour laquelle l'AME est calculé. Cette astuce peut être discutée, voir discutable lorsque l'AME est calculé pour une variable quanti.

Exemple (je ne sais pas si ces patterns sont dans l'échantillon, c'est juste pour l'exemple et j'espère ne pas avoir fait d'erreurs dans les calculs...manuels): 
effet marginal pour un homme de 18-24 ans, de niveau de formation = primaire et qui ne regarde pas la TV 
proba observée c'est un homme:  exp( -0.89005 + 0.41764)/(1+ exp( -0.89005 + 0.41764))=0.384
proba si c'était une femme: exp( -0.89005)/(1+ exp( -0.89005)) = 0.291
Effet marginal individuel: 0.384 - 0.291 = +0.093

Même chose pour l'âge et le niveau de formation, mais si la personne regarde la tv 10 heures par jour .
proba observée c'est un homme:  exp( -0.89005 + 0.41764 - 0.12093*10 )/(1+ exp( -0.89005 + 0.41764 - 0.12093*10)) = 0.157
proba si c'était une femme:  exp( -0.89005  - 0.12093*10 )/(1+ exp( -0.89005  - 0.1209 3*10)) = 0.109
Effet marginal individuel: 0.157 - 0.109 = +0.048

Ici, cela ne serait pas idiot de prendre le temps moyen d'exposition à la TV (2.25 hrs) qui est une valeur mesurée. Dans ce cas de figure, cette contribution individuelle à l'effet marginal pour les hommes serait de +0.087.

Je conseille vivement de lire la présentation de Cedric Afsa dans son document sur le modèle logit, surement le meilleur en fr (pages 43 à 46) =>https://www.insee.fr/fr/statistiques/fichier/2022139/Le-modele-Logit-CB.pdf).  
A lire également ses remarques sur l'utilisation abusive de l'expression "tout chose égale par ailleurs" lorsqu'on lit les résultats d'un modèle... mais c'est un tout autre sujet.

Très cordialement
MT

**Reply by larmarange** [reply]
Merci pour vos retours. Je ne connaissais pas le package marginaleffect et je vais me pencher dessus.

Ce chapitre est tout nouveau et encore en état de maturation. Merci pour votre feedback qui permettra de l améliorer.

**Reply by larmarange** [reply]
Bonjour

suite à vos conseils, le support de `marginaleffects` a été ajouté à `broom.helpers` et le chapitre a été entièrement réorganisé. C'est visible à https://larmarange.github.io/guide-R/analyses/estimations-marginales.html  

N'hésitez pas à faire un retour, des corrections sont toujours possibles.

Cdlt

**Comment by mthevenin** [comment]
Bonjour, 
Merci pour l'info je vais regardé cela. 

Et bienvenue à l'Ined comme chercheur associé. 
Bien cordialement 
marc 


De: "Joseph Larmarange" ***@***.***> 
À: "larmarange/guide-R" ***@***.***> 
Cc: "marc" ***@***.***>, "Comment" ***@***.***> 
Envoyé: Jeudi 12 Janvier 2023 12:29:33 
Objet: Re: [larmarange/guide-R] guide-R/analyses/effets-marginaux (Discussion #14) 





Bonjour 

suite à vos conseils, le support de marginaleffects a été ajouté à broom.helpers et le chapitre a été entièrement réorganisé. C'est visible à [ https://larmarange.github.io/guide-R/analyses/estimations-marginales.html | https://larmarange.github.io/guide-R/analyses/estimations-marginales.html ] 

N'hésitez pas à faire un retour, des corrections sont toujours possibles. 

Cdlt 

— 
Reply to this email directly, [ https://github.com/larmarange/guide-R/discussions/14#discussioncomment-4666684 | view it on GitHub ] , or [ https://github.com/notifications/unsubscribe-auth/AH7IIUSXN32ZVDH322GQ7PTWR7TJ3ANCNFSM6AAAAAATMA27FM | unsubscribe ] . 
You are receiving this because you commented. Message ID: ***@***.***> 


-- 
Marc Thevenin 
Service des Méthodes Statistiques 
Bureau 3.092 
Tel: 62107
